### PR TITLE
Feature/ccta/aembootcamp 06

### PR DESCRIPTION
**Pull request by cctorres**
Breadcrumb Component

Extended from Breadcrumb V1
- Breadcrumb links should be clickable and take the user to the respective page
when clicked.
- If hideInNav property is set to true, then page should not visible in breadcrumb
hierarchy.
- Breadcrumb should always have Homepage as first item.
- Fetch homepage dynamically with the help of homepage template

Breadcrumb

- Breadcrumb should not display in homepage
Breadcrumb Home

- Breadcrumb should prioritize navigation title then page title then jcr:title.
Breadcrumb preferences

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/structure/breadcrumb/.content.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="cq:Component"
+    jcr:title="Breadcrumb"
+    jcr:description="TThis is breadcrumb component for AEM./"
```
Comment (valMoncadaS) [inline ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/structure/breadcrumb/.content.xml]: Please fix description

File: ui.frontend/src/main/webpack/components/_breadcrumb.scss [hunk 2]
```diff
@@ -1,5 +1,20 @@
-.cmp-breadcrumb {}
-.cmp-breadcrumb__list {}
-.cmp-breadcrumb__item {}
-.cmp-breadcrumb__item--active {}
-.cmp-breadcrumb__item-link {}
\ No newline at end of file
+.cmp-breadcrumb__list {
```
Comment (valMoncadaS) [inline ui.frontend/src/main/webpack/components/_breadcrumb.scss:1]: please update styles. Current page item shouldn't be bold

**Review by valMoncadaS — CHANGES_REQUESTED** [review]

File: ui.frontend/src/main/webpack/components/_breadcrumb.scss (hunk 2 above)
Comment (cctorres) [reply ui.frontend/src/main/webpack/components/_breadcrumb.scss:1]: Screenshot 2024-05-30 at 9 22 09 AM

New  Commit will fix it

File: ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/structure/breadcrumb/breadcrumb.html
Comment (valMoncadaS) [inline ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/structure/breadcrumb/breadcrumb.html:1]: Where are you using hideInNav property?

File: core/src/main/java/com/aembootcamp/core/models/BreadcrumbModel.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Delegate @Self @Via(type = ResourceSuperType.class)
+    private Breadcrumb breadcrumb;
+
+    public boolean isHomePath() {
+        PageManager pageManager = resourceResolver.adaptTo(PageManager.class);
+        if (pageManager != null) {
+            Page currentPage = pageManager.getContainingPage(currentResource);
+            if (currentPage != null) {
+                String pagePath = currentPage.getPath();
```
Comment (lflondonol92) [inline core/src/main/java/com/aembootcamp/core/models/BreadcrumbModel.java]: Change this implementation. You can verify if the current page is the home page based on the Template name.

File: core/src/main/java/com/aembootcamp/core/models/BreadcrumbModel.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import org.apache.sling.models.annotations.via.ResourceSuperType;
+import lombok.experimental.Delegate;
+
+@Model(adaptables = SlingHttpServletRequest.class, adapters = Breadcrumb.class, resourceType = BreadcrumbModel.RESOURCE_TYPE)
+public class BreadcrumbModel implements Breadcrumb {
+
+    protected static final String RESOURCE_TYPE = "aem-bootcamp/components/structure/breadcrumb";
+
+    @SlingObject
+    private Resource currentResource;
```
Comment (lflondonol92) [inline core/src/main/java/com/aembootcamp/core/models/BreadcrumbModel.java]: Validate if you can change the currentResource with:

@SlingObject
private Page currentPage;

File: core/src/main/java/com/aembootcamp/core/models/BreadcrumbModel.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    @SlingObject
+    private ResourceResolver resourceResolver;
+
+    @Delegate @Self @Via(type = ResourceSuperType.class)
+    private Breadcrumb breadcrumb;
+
+    public boolean isHomePath() {
```
Comment (lflondonol92) [inline core/src/main/java/com/aembootcamp/core/models/BreadcrumbModel.java]: Rename this to isHomePage

File: ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/structure/breadcrumb/breadcrumb.html [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<sly data-sly-use.breadcrumb="com.aembootcamp.core.models.BreadcrumbModel" />
+<sly data-sly-test="${!breadcrumb.isHomePath}">
```
Comment (lflondonol92) [inline ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/structure/breadcrumb/breadcrumb.html]: change this to breadcrumb.homePage

reference: https://stackoverflow.com/questions/67930228/what-is-the-difference-between-using-is-or-get-in-java-when-calling-it-with

**Review by lflondonol92 — CHANGES_REQUESTED** [review]

**Review by lflondonol92 — APPROVED** [review]

**Pull request merged**